docs/@docs/label-rendering/text_renderer_class_diagram.mmd

```diff
@@ -0,0 +1,65 @@
+classDiagram
+
+    class TextElement
+    class TextElementGroup
+    class TextElementState
+    class TextElementGroupState
+    class TextElementStateCache
+    class TextElementsRenderer
+    class TextCanvasRenderer
+    class TextCanvas
+    class FontCatalog
+    class PoiRenderer
+    class PoiManager
+    class ScreenCollisions
+    class TextStyleCache
+    class Theme
+    class TextStyleDefinition
+    class TextRenderStyle
+    class TextLayoutStyle
+    class PoiRenderBuffer
+
+    TextElementsRenderer --> PoiManager
+    PoiManager --> PoiInfo: create
+    TextElementsRenderer --> ScreenCollisions
+    TextElementsRenderer --> "# font catalogs" TextCanvasRenderer
+    TextCanvasRenderer --> TextCanvas
+    TextCanvasRenderer --> PoiRenderer
```

Comment:
When trying in the live editor, I swapped lines 26 & 27, and it improved the diagram a bit, because the TextElement was closer to the GlyphData, so it didn't have to be spread so far.

Reply:
done.